### PR TITLE
Use Module#prepend only when it exists

### PR DESCRIPTION
**Pull request by RobinDaugherty**
[JRuby 1.7.x does not have a `Module#prepend` method](https://github.com/jruby/jruby/issues/751). #457 made the assumption that if Ruby version is 2.0 or higher, then the `prepend` method exists. This affected me because I'm [using JRuby 1.7 in "2.0 mode"](http://stackoverflow.com/a/15281071/1589422).

It's always better to use duck typing instead of assuming that a version of Ruby has specific methods.

I believe it's correct to treat a missing `prepend` method by using the pre-Ruby 2.0 method (`insert`ing `OldDebugExceptionsCatcher`), and it seems to work correctly. But it may not be the correct behavior.

### CODE REVIEW

**Comment by nateberkopec** [comment]
Can you rebase this against the latest `travis.yml`?

File: .travis.yml [hunk 1]
```diff
@@ -21,6 +21,19 @@ notifications:
     on_failure: change
 
 matrix:
+  include:
+    - rvm: jruby-1.7.24
+      env: JRUBY_OPTS="--2.0"
+      gemfile: gemfiles/rails41.gemfile
```
Comment (nateberkopec) [inline .travis.yml]: There is no rails41.gemfile on latest master, these need to be omitted.

**Comment by RobinDaugherty** [comment]
`LoadError: no such file to load -- raven`
Does that failure mean anything to you? I get the feeling that it's not related to a change that I made.

**Comment by nateberkopec** [comment]
Heh, no, I was hoping you would know :laughing:

**Comment by RobinDaugherty** [comment]
Doh! It makes sense...the gemfile doesn't exist, so raven doesn't exist in the bundle.

**Comment by Elektron1c97** [comment]
Wow, thanks for the fix, didn't know that something like JRuby 1.7 in 2.0 mode exists..
(I was the one who made the fix for the `Module#prepend` stuff)
Good work!

**Comment by RobinDaugherty** [comment]
Thanks @Elektron1c97. I hate to admit that the reason I know about it is that I'm using it in production!

**Comment by nateberkopec** [comment]
Merged in a4624495bf99c8da04f6

**Pull request closed without merging**